mobile/src/components/AllBooksWithFilters.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useState } from 'react';
+import { Text, View, StyleSheet, Dimensions, Pressable, Image } from 'react-native';
+
+import { LoadingCircle } from './LoadingCircle';
+import { PaginationBookList } from './PaginationBookList';
+import { useBookSearchFilter } from './BookSearchFilter';
+import { Book } from '../models/Book';
+import { TextStyles } from '../styles/TextStyles';
+import { Languages } from '../models/Languages';
+import { Colors } from '../styles/Colors';
+
+// how many books to display per page (in the pagination booklist)
+const booksPerPage = 9;
+
+const { width } = Dimensions.get('window');
+
+type AllBooksWithFiltersProps = { allBooks: Book[], loading: boolean};
+
+// ISO 639-1 2-letter language codes (from the Languages file) and the number of languages
+const langCodes = Object.keys(Languages);
+const numLangs = langCodes.length;
+
+// array of objects containing the language code and whether the language is "active" (if its button is pressed)
+// for use in the filter with default of "not active"
+const langArray = [];
+langCodes.forEach(el => {
+  langArray.push({ lang: el, isActive: false });
+});
+
+// object containing the array of language objects for use as a state
+const langState = {
+  arr: langArray,
+};
```

Comment:
This is the problem with the checkboxes, these need to be proper state variables inside components. I would add a Set of selected languages to the filter hook, like:

```ts
const [selectedLangs, setSelectedLangs] = useState(new Set<Language>());

// on select lang:
setSelectedLangs(langs => new Set(langs).add('en'));

// on deselect
setSelectedLangs(langs => new Set(langs).delete('en'));
```